imports/server/methods/createPuzzle.ts

```diff
@@ -81,7 +91,39 @@ defineMethod(createPuzzle, {
       await ensureDocument(fullPuzzle, docType);
     }
 
-    await Puzzles.insertAsync(fullPuzzle);
+    // In a transaction, look for a puzzle with the same URL. If present, we
+    // reject the insertion unless the client overrides it.
+    const client = MongoInternals.defaultRemoteCollectionDriver().mongo.client;
+    const session = client.startSession();
+    try {
+      await session.withTransaction(async () => {
+        if (url) {
+          const existingPuzzleWithUrl = await Puzzles.collection
```

Comment:
I think it would be good to **also** do this sort of check **before** the `ensureDocument` call above -- in the case where there is already another puzzle with the same URL (i.e. the operator missed that the puzzle had already been entered), we can reject the request quickly and without causing side-effects or unnecessary Google Drive API calls (which are often slow relative to our internal method invocations).